### DISCUSSION TITLE
Adding `Model::whereEmpty('column')` and `Model::whereNotEmpty('column')` to the query builder

### DISCUSSION THREAD

**Discussion by kylemilloy** [laravel/framework#47985] [Ideas]
I have an eloquent model where sometimes the value is `NULL` and sometimes it's an empty string and with the DB I'm using I need to use the following to get both values:

```php
Model::whereNull('column')
  ->orWhere('column', '')
```

What I think might be a nice helper function is where we check for `whereEmpty('column')` instead. My idea here is that this follows PHP's `empty($var)` method and behind the scenes would evaluate similarly. 

```php
// if `column` is NULL, false, 0, '', or [] it will be returned
Model::whereEmpty('column')

// if `column` is NOT NULL, false, 0, '', or [] it will be returned
Model::whereNotEmpty('column')
```

I'm not wild about including JSON/arrays into this as that might have other side effects (like shouldn't {} also be empty?) but am looking to bounce some ideas off people and can add this. 

Similar discussion [here](https://github.com/laravel/framework/discussions/45130) but looks like it never got traction. 

### Edit

Might also be good to include `orWhereEmpty` and `orWhereNotEmpty`? But maybe that's a yagni situation.

**Comment by hypnodev** [comment]
To be honest would make sense, but isn't more safe to review the place where field is updated and force null in case is an empty string?

**Reply by kylemilloy** [reply]
In the scope of an application, you're right: it's safer (and also probably more performant) to sanitize the data before insert so that I'm only having to search for `NULL`. 

The one hitch I have is that I do not have total control over the data as I'm inheriting a legacy DB that has multiplpe values (`null` and `''`...and even `'undefined'` 🙄) that I need to account for. Going forward the plan would be to replace the endpoint that inserts data with something that will prevent this but at this time that feature is out of scope.
